lib/rules/headers/shortname.js

```diff
@@ -0,0 +1,204 @@
+                    var hasPreviousVersion = !(await sr.isFP());
+                    if (hasPreviousVersion && historyStatusCode === 404) {
+                        // it's a none FP spec, but the history page doesn't exist. There should be a 'valid' previous-shortname.
+                        const previousShortname = linkHistory.getAttribute(
+                            'data-previous-shortname'
+                        );
+                        if (previousShortname) {
+                            // prettier-ignore
+                            sr.warning(historyError, 'this-previous-shortname',
+                                {
+                                    previousShortname,
+                                    thisShortname: shortname,
+                                }
+                            );
+
+                            // check previous shortname is valid.
+                            const previousHistoryHref = `https://www.w3.org/standards/history/${previousShortname}`;
+                            let previousHistoryStatusCode;
+                            try {
+                                const res = await doAsync(superagent).head(
+                                    previousHistoryHref
+                                );
+                                previousHistoryStatusCode = res.statusCode;
+                            } catch (err) {
+                                previousHistoryStatusCode = err.status;
+                            }
+
+                            if (previousHistoryStatusCode === 404) {
+                                sr.error(historyError, 'history-bad-previous', {
+                                    previousShortname,
+                                    url: previousHistoryHref,
+                                });
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (dts.Rescinds) {
+        const linkRescinds = dts.Rescinds.dd.querySelector('a');
+
+        if (linkRescinds && linkRescinds.href) {
+            matches = (linkRescinds.getAttribute('href') || '')
+                .trim()
+                .match(
+                    /^https:\/\/www\.w3\.org\/TR\/\d{4}\/REC-(.+)-\d{8}\/?$/
+                );
+            if (matches) {
+                sn = matches[1];
+                if (sn !== shortname)
+                    sr.error(self, 'this-rescinds-shortname', {
+                        rescindsShortname: sn,
+                        thisShortname: shortname,
+                    });
+            }
+        }
+    }
+
+    // check shortname is valid.
+    const isFP = await sr.isFP();
+    // FP documents cannot use existing shortname
+    if (
+        sr.config.longStatus === 'First Public Working Draft' &&
+        !isFP &&
+        shortname
+    ) {
+        sr.error(self, 'shortname-existed', {
+            status: sr.config.longStatus,
+        });
```

Comment:
changes related to the suggestion in `l10-en_GB.js`.
```suggestion
        sr.error(self, 'shortname-existed');
```